client/quantum_serverless/core/provider.py

```diff
@@ -308,8 +308,18 @@ def widget(self):
         return Widget(self).show()
 
 
-class Provider(BaseProvider):
-    """Provider."""
+class ServerlessProvider(BaseProvider):
```

Comment:
We need to deprecate it first before removing, since it is one of the most used class.

Something like
```python
class ServerlessProvider(BaseProvider):
    ...

class Provider(ServerlessProvider):
    def __init__(....):
         warnings.warn(
              "`Provider` class has been deprecated and will be removed in release 0.8.0. Please, use `ServerlessProvider` instead.", 
              DeprecationWarning)
         self.super(...)

```

Reply:
Will do